### ISSUE TITLE
Inconsistent behaviour ios/jvm, write() doesn't update position on iOS

### ISSUE THREAD

**Issue by jbruchanov** [DitchOoM/buffer#49]
Using version 1.3.9

Having following test
```
@Test
fun test() {
    val text = "Hello world!"
    val input = PlatformBuffer.wrap(text.encodeToByteArray())
    val output = PlatformBuffer.allocate(text.length)
    output.write(input)
    assertThat(input.position()).isEqualTo(text.length) //<- ios fails
    assertThat(output.position()).isEqualTo(text.length)
    output.position(0)
    assertThat(output.readString(text.length)).isEqualTo(text)
}
```
JVM: OK
iOS: fails
```
com.willowtreeapps.opentest4k.AssertionFailedError: expected:<[12]> but was:<[0]>
Expected :[12]
Actual   :[0]
```

**Comment by thebehera** [comment]
Interesting, that is definitely a bug. I should be able to get a fix up within a day or two. Thanks for filling this issue!

**Comment by thebehera** [comment]
should be fixed in 1.3.10
